From 12db1dc2a8ac3d5cf8db000ef89e3ec7ad27aa1a Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Tue, 24 Sep 2013 18:05:41 +0200
Subject: [PATCH] Use dbus system bus instead of session bus for dbus
 communication

We're not using user sessions in webOS so we have to run every dbus service on the system
service.

Upstream-Status: Inappropiate [other]

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 connection/inputcontextdbusaddress.cpp | 2 +-
 connection/serverdbusaddress.cpp       | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/connection/inputcontextdbusaddress.cpp b/connection/inputcontextdbusaddress.cpp
index 306f870..43ac288 100644
--- a/connection/inputcontextdbusaddress.cpp
+++ b/connection/inputcontextdbusaddress.cpp
@@ -51,7 +51,7 @@ void DynamicAddress::get()
                                                           DBusPropertiesInterface, DBusPropertiesGetMethod);
     message.setArguments(arguments);
 
-    QDBusConnection::sessionBus().callWithCallback(message, this,
+    QDBusConnection::systemBus().callWithCallback(message, this,
                                                    SLOT(successCallback(QDBusVariant)),
                                                    SLOT(errorCallback(QDBusError)));
 }
diff --git a/connection/serverdbusaddress.cpp b/connection/serverdbusaddress.cpp
index b80a174..c078cb6 100644
--- a/connection/serverdbusaddress.cpp
+++ b/connection/serverdbusaddress.cpp
@@ -34,8 +34,8 @@ AddressPublisher::AddressPublisher(const QString &address)
     : QObject()
     , mAddress(address)
 {
-    QDBusConnection::sessionBus().registerObject(MaliitServerObjectPath, this, QDBusConnection::ExportAllProperties);
-    if (!QDBusConnection::sessionBus().registerService(MaliitServerName)) {
+    QDBusConnection::systemBus().registerObject(MaliitServerObjectPath, this, QDBusConnection::ExportAllProperties);
+    if (!QDBusConnection::systemBus().registerService(MaliitServerName)) {
         qWarning("maliit-server is already running");
         std::exit(0);
     }
@@ -43,7 +43,7 @@ AddressPublisher::AddressPublisher(const QString &address)
 
 AddressPublisher::~AddressPublisher()
 {
-    QDBusConnection::sessionBus().unregisterObject(MaliitServerObjectPath);
+    QDBusConnection::systemBus().unregisterObject(MaliitServerObjectPath);
 }
 
 QString AddressPublisher::address() const
-- 
1.8.1.2

